Core/Utilities/TimeLeft/TimeLeft.py

```diff
@@ -76,10 +85,23 @@ def getTimeLeft( self, cpuConsumed = 0.0 ):
       return resourceDict
 
     resources = resourceDict['Value']
-    if not resources['CPULimit'] or not resources['WallClockLimit']:
+    self.log.verbose( resources )
+    if not resources['CPULimit'] and not resources['WallClockLimit']:
       # This should never happen
       return S_ERROR( 'No CPU or WallClock limit obtained' )
 
+    # if one of CPULimit or WallClockLimit is missing, compute a reasonable value
+    if not resources['CPULimit']:
+      resources['CPULimit'] = resources['WallClockLimit']
```

Comment:
This is not good: we should set a CPULimit lower than WallClock limit, although we don't know the ratio,  Iwould propose a factor 0.8 or 0.7
Similarly if wall clock limit is not set, we can set it to CPULimit / 0.8 in order to be only limited by CPULimit